src/EnergyPlus/IceThermalStorage.hh

```diff
@@ -54,120 +54,60 @@
-
-    // MODULE VARIABLE DECLARATIONS:
-    extern bool ResetXForITSFlag;
-
-    // Input data
-    extern Real64 ITSNomCap;  // Design nominal capacity of Ice Thermal Storage [J] (user input in GJ)
-    extern int InletNodeNum;  // Node number on the inlet side of the plant
-    extern int OutletNodeNum; // Node number on the inlet side of the plant
-
     // ITS numbers and FoundOrNot
-    extern int IceNum;
-    extern int NumIceStorages;
-    extern bool IceStorageNotFound;
-    extern int NumDetIceStorages;
-    extern int TotalIceStorages;
-    // ITS UAice and HLoss
-    extern Real64 UAIceCh;    // Charging Ice Thermal Storage overall heat transfer coefficient [W/C]
-    extern Real64 UAIceDisCh; // Discharging Ice Thermal Storage overall heat transfer coefficient [W/C]
-    extern Real64 HLoss;      // ITS Heat Loss
-    // ITS State
-    extern Real64 XCurIceFrac; // Current Fraction of Ice Thermal Storage remaining [fraction]
-    extern Real64 U;           // Adjusted input U after reading U Schedule [fraction]
-    extern Real64 Urate;       // Final Urate adjusted Urate based on Error protection (I) [fraction] by HOUR
-    // ITS status information
-    extern Real64 ITSMassFlowRate;       // ITS water mass flow rate [kg/s]
-    extern Real64 ITSInletTemp;          // ITS inlet water temperature [C]
-    extern Real64 ITSOutletTemp;         // ITS outlet water temperature [C]
-    extern Real64 ITSOutletSetPointTemp; // ITS outlet water temperature setpoint [C]
-    extern Real64 ITSCoolingRate;        // ITS Discharge(-)/Charge(+) rate [W]
-    extern Real64 ITSCoolingEnergy;
-    extern Real64 ChillerOutletTemp; // Chiller outlet brine temperature [C]
-    extern Array1D_bool CheckEquipName;
-
-    // SUBROUTINE SPECIFICATIONS FOR MODULE
-    // General routine
-
-    // Types
-
-    struct IceStorageMapping
+    extern int NumSimpleIceStorage;
+    extern int NumDetailedIceStorage;
+    extern int TotalNumIceStorage;
+
+    struct IceStorageType
     {
-        // Members
-        // Input data
-        std::string Name; // User identifier
-        std::string StorageType;
-        int StorageType_Num;
-        int LocalEqNum;
+        enum
+        {
+            Simple,
+            Detailed
+        };
+    };
 
-        // Default Constructor
-        IceStorageMapping() : StorageType_Num(0), LocalEqNum(0)
+    struct CurveVars
+    {
+        enum
         {
-        }
+            FracChargedLMTD,
+            FracDischargedLMTD,
+            LMTDMassFlow,
+            LMTDFracCharged
+        };
     };
 
-    struct IceStorageSpecs
+    struct DetIce
+    {
+        enum
+        {
+            InsideMelt, // Inside melt system--charge starting with bare coil
+            OutsideMelt // Outside melt system--charge from existing ice layer on coil
+        };
+    };
+
+    struct ITSType
+    {
+        enum
+        {
+            IceOnCoilInternal,
+            IceOnCoilExternal
+        };
+    };
```

Comment:
The enum inside a struct is a bit odd.  Are you able to just use a standard `enum class ITSType` for these?

Reply:
I'll fix it up.